src/main/java/technology/semi/weaviate/client/base/http/impl/CommonsHttpClientImpl.java

```diff
@@ -1,33 +1,31 @@
 package technology.semi.weaviate.client.base.http.impl;
 
 import org.apache.http.HttpHeaders;
-import org.apache.http.client.methods.CloseableHttpResponse;
-import org.apache.http.client.methods.HttpDelete;
-import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpHead;
-import org.apache.http.client.methods.HttpPatch;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.methods.HttpPut;
-import org.apache.http.client.methods.HttpRequestBase;
-import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.*;
 import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.util.EntityUtils;
 import technology.semi.weaviate.client.base.http.HttpClient;
 import technology.semi.weaviate.client.base.http.HttpResponse;
 
+import java.io.Closeable;
+import java.io.IOException;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.util.Map;
 
-public class CommonsHttpClientImpl implements HttpClient {
+public class CommonsHttpClientImpl implements HttpClient, Closeable {
   private final Map<String, String> headers;
-  private final CloseableHttpClientBuilder clientBuilder;
+  private final CloseableHttpClient client;
 
   public CommonsHttpClientImpl(Map<String, String> headers, CloseableHttpClientBuilder clientBuilder) {
     this.headers = headers;
-    this.clientBuilder = clientBuilder;
+    this.client = clientBuilder.build();
+  }
+
+  @Override
+  public void close() throws IOException {
```

Comment:
`close()` is never used. 
`CommonsHttpClientImpl` is part of `BaseClient` (and its child classes), but `BaseClient` gives no access to `close` method to the user